content/documentation/pages/5-feature-guides/1-streams/7-scaling.md

```diff
@@ -0,0 +1,33 @@
+The ability of a streaming pipeline to cope with increased data load depends on the following characteristics:
+
+- _Messaging Middleware_ - [data partitioning](https://docs.spring.io/spring-cloud-stream/docs/current/reference/htmlsingle/#partitioning) is a widely used technique for scaling the messaging middleware infrastructure. Spring Cloud Data Flow through Spring Cloud Stream, provides excellent support for [streaming partitioning](%currentPath%/feature-guides/streams/partitioning/#data-partitioning/).
+
+- _Event-driven Applications_ - Spring Cloud Stream provides support for data processing in parallel with multiple consumer instances, which is commonly referred to as application scaling.
+
+Following diagram illustrates a typical scale-out architecture based on data partitioning and application parallelism:
+
+![SCDF Stream Scaling](images/scdf-stream-partition-and-scaling-overview.png)
+
+Platforms such as Kubernetes and Cloud Foundry offer scaling features for the Operators to control system’s load. For example, Operators can leverage [cf scale](https://docs.cloudfoundry.org/devguide/deploy-apps/cf-scale.html) to [scale applications](https://docs.spring.io/spring-cloud-dataflow/docs/2.3.0.RELEASE/reference/htmlsingle/#configuration-cloudfoundry-scaling) in Cloud Foundry and likewise, use the [kubectl scale](https://jamesdefabia.github.io/docs/user-guide/kubectl/kubectl_scale/) to scale applications in Kubernetes. It is also possible to enable autoscaling features with the help of [App Autoscaler](https://docs.run.pivotal.io/appsman-services/autoscaler/using-autoscaler.html) in Cloud Foundry and [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) or [KEDA](https://github.com/kedacore/keda) in Kubernetes respectively. The autoscaling is usually determined by the CPU or memory limit thresholds or message queue-depth and topic-offset-lag metrics.
+
+While the scaling happens outside of the Spring Cloud Data Flow, the applications that are scaled can react to and handle the upstream load automatically. Developers only need to configure the message partitioning using properties such as [partitionKeyExpression](https://cloud.spring.io/spring-cloud-static/spring-cloud-stream/3.0.0.RELEASE/reference/html/spring-cloud-stream.html#spring-cloud-stream-overview-configuring-output-bindings-partitioning) and [partitionCount](https://cloud.spring.io/spring-cloud-static/spring-cloud-stream/3.0.0.RELEASE/reference/html/spring-cloud-stream.html#spring-cloud-stream-overview-configuring-output-bindings-partitioning).
+
+In addition to the platform specific, low-level APIs, Spring Cloud Data Flow provides a dedicated, Scale API, designed for scaling data pipelines. The Scale API provides a uniform way to leverage the platform scaling capabilities from within custom Spring applications.
```

Comment:
> The Scale API provides a uniform way to leverage the platform scaling capabilities from within custom Spring applications.

Not sure I follow this. Esp. the "from within custom Spring applications". It makes it sound that the scaling is happening inside the custom applications.

Reply:
I see you point. My intention is to stress that the Scale API still uses (internally) the native scaling capabilities of the different platforms but exposes in a single, unified interface. Perhaps something like?
> The Scale API unifies the various platforms native scaling capabilities into a uniform and simple interface.